ingestion/src/metadata/ingestion/source/dashboard/powerbi/connection.py

```diff
@@ -22,28 +22,34 @@
 )
 from metadata.ingestion.connections.test_connections import test_connection_steps
 from metadata.ingestion.ometa.ometa_api import OpenMetadata
-from metadata.ingestion.source.dashboard.powerbi.client import PowerBiApiClient
+from metadata.ingestion.source.dashboard.powerbi.client import (
+    PowerBiApiClient,
+    PowerBiClient,
+)
+from metadata.ingestion.source.dashboard.powerbi.file_client import PowerBiFileClient
 
 
 def get_connection(connection: PowerBIConnection) -> PowerBiApiClient:
     """
     Create connection
     """
-    return PowerBiApiClient(connection)
+    client = PowerBiClient(api_client=PowerBiApiClient(connection))
+    if connection.pbitFilesSource:
+        client.file_client = PowerBiFileClient(connection)
```

Comment:
Doing it this way defeats a little bit the purpose of Pydantic since you'll have no validation whatsoever when assigning it directly.

(I think it's fine on this case but wanted to give you a heads-up

Reply:
Makes sense. I just wanted to wrap the clients in a class so that they can be accessed wherever easily

Reply:
I think the wrapping is fine to be honest, maybe shouldn't be a pydantic class or you could do something like:
```
file_client = None
if connection.pbitFilesSource:
  file_client = PowerBiFileClient(connection)
# Instantiate pydantic class with both clients (or one None)
```

Reply:
Got it will make the change

Reply:
This is updated